src/app/guard/auth.guard.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import {
+  Router,
+  CanActivate,
+  ActivatedRouteSnapshot,
+  RouterStateSnapshot,
+} from '@angular/router';
+import { select, Store } from '@ngrx/store';
+import { IUser } from '../model/user.model';
+import { selectUser } from '../store/selector/auth.selector';
+import { IAppState } from '../store/state/app.state';
+
+@Injectable({ providedIn: 'root' })
+export class AuthGuard implements CanActivate {
+  user?: IUser;
+
+  constructor(
+    private store: Store<IAppState>,
+    private router: Router
+  ) {
+    this.store.pipe(select(selectUser)).subscribe((user) => (this.user = user));
+  }
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+    if (this.user) {
```

Comment:
canActivate can return an observable, get the latest user in the method and do not keep it as a state

Reasoning: there may be the case that it takes some time until user variable is refreshed, and canActivate might return true in the meantime

Reply:
Also try to add types to method returns always